repo/content/indexblob/index_blob.go

```diff
@@ -8,13 +8,14 @@ import (
 	"github.com/kopia/kopia/internal/gather"
 	"github.com/kopia/kopia/repo/blob"
 	"github.com/kopia/kopia/repo/content/index"
+	"github.com/kopia/kopia/repo/maintenancestats"
 )
 
 // Manager is the API of index blob manager as used by content manager.
 type Manager interface {
 	WriteIndexBlobs(ctx context.Context, data []gather.Bytes, suffix blob.ID) ([]blob.Metadata, error)
 	ListActiveIndexBlobs(ctx context.Context) ([]Metadata, time.Time, error)
-	Compact(ctx context.Context, opts CompactOptions) error
+	Compact(ctx context.Context, opts CompactOptions) (*maintenancestats.DropDeletedIndexesStats, error)
```

Comment:
`indexblob.Manager.Compact` is used for two conceptually distinct and separate tasks:
- dropping deleted contents;
- legacy index compaction (V0 indexes only)

The `Compact` function has different implementations depending on the index manager, albeit the same function signature.

Instead of returning stats from the `Compact` function, it should be enough to return the stats from `runTaskDropDeletedContentsFull`
That would simplify this change.

Reply:
See discussions in https://github.com/kopia/kopia/pull/4948#discussion_r2487692809